src/eigen/preconditioners.jl

```diff
@@ -68,4 +70,11 @@ end
 
 function precondprep!(P::PreconditionerTPA, X)
     P.mean_kin = [real(dot(x, Diagonal(P.kin), x)) for x in eachcol(X)]
```

Comment:
Interesting, what do you get if you map over eachcol(::GPUArray)? An Array or a GPUArray?

Reply:
We get a GPUArray. The is the same type of problem as the issue you raised [here](https://github.com/JuliaMolSim/DFTK.jl/pull/697#discussion_r929222161). List comprehension is really handy for CPU but sometimes can't be directly used when building GPU Arrays.